subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheStoreBuildOperationType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal;
+
+import org.gradle.caching.internal.controller.BuildCacheServiceRole;
+import org.gradle.internal.operations.BuildOperationType;
+import org.gradle.internal.scan.UsedByScanPlugin;
+
+/**
+ * A store operation to a build cache.
+ *
+ * A store operation may STORE or NOT-STORE.
```

Comment:
The equivalent in the load is a miss. There's no guarantee that the build cache service does anything when asked to store. Also, it really should be its decision on whether it wants to write to its backing. Granted, there's nothing for it to really make this decision on now, but its easy to imagine a service not wanting to store an artifact over a certain size.

That said, I'm happy to make it an error condition if the service doesn't actual read the data. I'll make that change speculatively.